### PR TITLE
manage some missing actions of entity rules

### PR DESCRIPTION
**Pull request by orthagh**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #11338

Pretty rough fix atm, just to validate the issue/fix
I don't know, yet, how to do a more generic way to affect possible actions of ruleentity.
We may need a test set also

One last thing, i think maybe we missed the action `_affect_entity_by_tag`
This needs to be confirmed

### CODE REVIEW

**Comment by cedric-anne** [comment]
@orthagh Could you add some tests ?

**Comment by orthagh** [comment]
> @orthagh Could you add some tests ?

done

File: src/Inventory/Asset/MainAsset.php [hunk 1]
```diff
@@ -595,8 +609,10 @@ public function rulepassed($items_id, $itemtype, $rules_id, $ports_id = 0)
         $val->is_dynamic = 1;
         $val->entities_id = $entities_id;
 
-        $val->states_id = $this->states_id_default ?? $this->item->fields['states_id'] ?? 0;
```
Comment (trasher) [inline src/Inventory/Asset/MainAsset.php:598]: Are default configured values automatically used already?

File: tests/functionnal/RuleImportEntity.php [hunk 2]
```diff
@@ -220,4 +230,89 @@ public function testRefusedByEntity()
...
+        $location = new \Location();
+        $location_id = $location->add([
+            'name' => 'Location 1'
+        ]);
+        $this->integer($location_id)->isGreaterThan(0);
+
+        $group = new \Group();
+        $group_id = $group->add([
+            'name' => 'Group tech 1'
+        ]);
```
Comment (trasher) [inline tests/functionnal/RuleImportEntity.php:252]:
```suggestion
        ]);
        $this->integer($group_id)->isGreaterThan(0);
```

File: tests/functionnal/RuleImportEntity.php [hunk 3]
```diff
@@ -220,4 +230,89 @@ public function testRefusedByEntity()
...
+        $this->integer($ruleaction->add($input))->isGreaterThan(0);
+        $input = [
+            'rules_id'    => $rule_id,
+            'action_type' => 'assign',
+            'field'       => 'users_id_tech',
+            'value'       => $user_id
+        ];
+        $this->integer($ruleaction->add($input))->isGreaterThan(0);
+
+
```
Comment (trasher) [inline tests/functionnal/RuleImportEntity.php:303]:
```suggestion
```

File: src/Inventory/Asset/MainAsset.php (hunk 1 above)
Comment (orthagh) [reply src/Inventory/Asset/MainAsset.php:598]: you're right, missed these

**Review by trasher — APPROVED** [review]

File: tests/functionnal/RuleImportEntity.php [hunk 4]
```diff
@@ -220,4 +230,89 @@ public function testRefusedByEntity()
         ];
         $this->array($ent)->isEqualTo($expected);
     }
+
+    /**
+     * We want to test optional actions provided by ruleentit, like:
+     * - location
+     * - location
```
Comment (cedric-anne) [inline tests/functionnal/RuleImportEntity.php:237]:
```suggestion
     * We want to test optional actions provided by RuleImportEntity, like:
     * - location assignment
     * - group assignment
     * - ...
```

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**